### ISSUE TITLE
[FEATURE_REQUEST] Limited user mode (locked-down scenario-specific functionality)

### ISSUE THREAD

**Issue by Denubis** [SillyTavern/SillyTavern#2729]
### Have you searched for similar requests?

Yes

### Is your feature request related to a problem? If so, please describe.

This builds on #2227 -- but we've found remarkable success using this in our classrooms. The problem is that the usermode defaults are able to use ST as users (as intended) which means they can control the scenario settings, swap LLM providers (tediously expensive when we disovered a few of our students had done this) and delete old chats. 

### Describe the solution you'd like

As educators, we would like to set groups of 700 students a scenario, provide a URL, and expect that they will engage with the scenario -- without the ability for them to swap to other scenarios. We would like a "limited user" mode where a user logging in will be presented a new chat, as a specific persona, and with a specific character that chat is associated with, but with the menu bar disabled, and with some functions of the options button disabled.

(As a quality of life thing, it'd be amazing to be able to bulk-enroll folks via API call or .json file. I had to write a chromium script to make all 350 tutorial pairs. But I solved this, so it's "fine") 

### Describe alternatives you've considered

Looking at the code, specifically: https://github.com/SillyTavern/SillyTavern/blob/staging/default/user.css I suppose that I can set custom css to disable/hide HTML elements. The main thing I'm going to have to figure out is how to get a specific character selected by default. This won't solve the new-chat issue, but is a workaround we can play with. 

I'd feel a lot happier if these capabilities were denied by basic ACL rather than client-side CSS. It would allow us to go beyond monitored-tutorial mode in terms of offering these experiences to our students.

### Additional context

We might have some budget available to commission development for features related to this request. If there's an extension author you would recommend to us, let me know and I'll hop on your discord. (I'm not offering donations to you as per the readme, but am happy to have other discussions.)

### Priority

Medium (Would be very useful)

### Are you willing to test this on staging/unstable branch if this is implemented?

Yes

**Comment by Wolfsblvt** [comment]
@Cohee1207 is boss about server structure/functionality.
But I *think* this is likely an "out of scope" feature request, as ST itself was never intended to be used this way. Similarly to how ST will never really be able to be used as a backend.

The introduction of user accounts for the server was mostly based on many users using different *environments* for ST, and having to use separate instances of the whole ST itself. Users will have different settings, characters, history, etc.
You can use this as one physical person - like for example many modern browsers (Chrome & co) have user profiles to separate your environment. Streaming services, like Netflix and Co have this too.
User accounts don't have rights or restrictions, besides the simplest form of access limitation via passwords.

Hiding elements in CSS or on client side is a small deterrent, but will not fix the issue. You can still use slash commands in chat for example to change the api or model, which you want to prevent. `/model gpt-4-turbo`, it would be.
Even if you somehow prevent slash command usage, people will be able to use browser dev tools to execute scripts directly on the ST client. They can install their own extensions, that do whatever they want. Even if you block the extension UI, the code to install extensions could likely be run via a simple Javascript. As ST is designed to allow other extensions/scripts to tie into many of its core functions.

As long as ST is a client-heavy application (which it is), smart kids will find a way to get around every restriction you might want to implement or add. Worst case, you can just dynamically change the client scripts you have loaded in the browser by setting breakpoints, modifying the files manually, rewriting values, and more.

If bandaid solutions might be enough and you only want to deter, not completely prevent, there might be way to block the most common things via an extension or even a fork for you, if you find a willing person to do this.

> We might have some budget available to commission development for features related to this request. If there's an extension author you would recommend to us, let me know and I'll hop on your discord. (I'm not offering donations to you as per the readme, but am happy to have other discussions.)

We have a "content requests" channel on the Discord, that some have already used to request and talked about extensions, or you could bring that up in one of the channels like "advanced features", which would likely fit.
Can't speak for the community coders really. If you want to find out, you can only try yourself. It's all hobby devs here.

**Comment by Denubis** [comment]
Yeah, I only need to deter -- this is to roadblock the curious, not prevent the malicious. I'll hop on discord. Thanks.

**Comment by Cohee1207** [comment]
Agree with this being out of scope for the application. Anything that limits the functionality and is impossible to disable (unlike simple UI mode, for example) is not expected to be included in the main package. Sidenote, that user accounts are probably not a proper name in retrospect, user profiles would be more appropriate.

**Comment by Wolfsblvt** [comment]
> Sidenote, that user accounts are probably not a proper name in retrospect, user profiles would be more appropriate.

Didn't want to say it outright in my message, but I agree.
This is more like Netflix/Chrome profiles. Not Windows user accounts.
